src/python/module/smartredis/util.py

```diff
@@ -111,3 +114,136 @@ def typecheck(arg, name, _type):
     """
     if not isinstance(arg, _type):
         raise TypeError(f"Argument {name} is of type {type(arg)}, not {_type}")
+
+
+
+class DatasetConverter:   
+    
+    def add_metadata_for_xarray(ds,data_names,dim_names,coord_names=None,attr_names=None):
+        """Extracts metadata from a SmartRedis dataset and adds it to
+        dataformat specific fieldnames 
+
+        :param ds: a SmartRedis Dataset with metadata added to Xarray 
+        specific field names 
+        :type ds: SmartRedis Dataset
+        :param data_names: variable data field name
+        :type data_names: list or string 
+        :param dim_names: dimension field names. Defaults to None.
+        :type dim_names: list
+        :param coord_names: coordinate field names. Defaults to None.
+        :type coord_names: list, optional
+        :param attr_names: attribute field names. Defaults to None.
+        :type attr_names: list, optional
+        """        
+        
+        args = ['dim_names','coord_names','attr_names'] 
+
+        for d in data_names:  
+
+            ds.add_meta_string("_xarray_data_names",d)
+   
+            
+            for arg in args:
+
+                if isinstance(eval(arg),list):
+                    items = []
+                    for item in eval(arg): 
+                        items.append(item)
+
+                    arg_field=",".join(items)
+                    ds.add_meta_string(f"_xarray_{d}_{arg}",arg_field)
+            
+                else:
+                    if eval(arg) != None: 
+                        ds.add_meta_string(f"_xarray_{d}_{arg}",eval(arg))
+                        print(f"_xarray_{d}_{arg}")
+    
+                    else: 
+                        ds.add_meta_string(f"_xarray_{d}_{arg}","null")
+    
+        
+     
+    def transform_to_xarray(ds): 
+        """ Uses dataformat specifc field names added to the 
+        SmartRedis dataset in the add_metadata method to construct 
+        xarray in transform method 
+
+        :param ds: a SmartRedis dataset that has had  metadata added 
+        to dataformat specificfield names 
+        :type ds: SmartRedis Dataset
+        
+        :return: a dictionary of keys as the data field anme and the
+        value as the built Xarray DataArray constructed using 
+        fieldnames and appropriately formatted metadata. 
+        """
+        
+        coord_dict= {}
+        coord_final = {}
+        attr_dict = {}
+        
+        
+        iter_ = ds.get_meta_strings("_xarray_data_names")
+  
+        for item in ds.get_meta_strings("_xarray_data_names"):
```

Comment:
In general, it might be good to use more descriptive variable names other than ``iter`` and ``item`` or one letter variables.  It makes the code hard to read.  Can we come up with better, more descriptive variable names?